### PR TITLE
Print git description and hashes consistently in CMake

### LINKED ISSUE #2646: LibraryVersions.txt contains the wrong hash
```
SpECTRE hash: v2020.12.07
```
That is (a) not a hash and (b) not a description of the commit I am on (with a clean run of cmake): 
```
$ git rev-parse HEAD v2020.12.07
b3174f5633fa9692eb069a2275df72f8bd516d80
296ac0428eb0191d79690b9f935918fdb0eac569
```

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Fixes #2646

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: cmake/CodeCoverage.cmake [hunk 1]
```diff
@@ -118,7 +113,7 @@ function(SETUP_TARGET_FOR_COVERAGE
       --output-file ${OUTPUT}.filtered.info
       # Generate HTML report
       COMMAND ${GENHTML} --legend --demangle-cpp
-      --title `cd ${CMAKE_SOURCE_DIR} && git rev-parse HEAD`
+      --title `cd ${CMAKE_SOURCE_DIR} && ${GIT_HASH}`
```
Comment (wthrowe) [inline cmake/CodeCoverage.cmake]: GIT_HASH is not a command.

File: cmake/SetupDoxygen.cmake [hunk 2]
```diff
@@ -149,7 +149,7 @@ ${DOCS_POST_PROCESS_COMMAND} && exit \${generate_docs_exit}\n"
       ${GENHTML} --legend
       --no-function-coverage
       --no-branch-coverage
-      --title `cd ${CMAKE_SOURCE_DIR} && git rev-parse HEAD`
+      --title `cd ${CMAKE_SOURCE_DIR} && ${GIT_HASH}`
```
Comment (wthrowe) [inline cmake/SetupDoxygen.cmake]: GIT_HASH is not a command.

**Review by wthrowe — CHANGES_REQUESTED** [review]
I have verified that this fixes #2646.

**Comment by nilsvu** [comment]
Also I missed a few occurrences in `WrapExecutableLinker.sh`, will fix tmrw

**Comment by nilsvu** [comment]
Done, please check the fixup @wthrowe

**Review by nilsdeppe — DISMISSED** [review]

**Comment by wthrowe** [comment]
Looks good.  Squash.

**Review by nilsdeppe — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**